rust/main/agents/relayer/src/msg/metadata/aggregation.rs

```diff
@@ -199,10 +199,24 @@ impl MetadataBuilder for AggregationIsmMetadataBuilder {
         }))
         .await;
 
-        // If any inner ISMs are refusing to build metadata, we propagate just the first refusal.
+        // If any inner ISMs are refusing to build metadata,
+        // we propagate just the first refusal.
         for sub_module_res in sub_modules_and_metas.iter() {
+            // This occurs when there is a reorg
             if let Err(MetadataBuildError::Refused(s)) = sub_module_res {
-                return Err(MetadataBuildError::Refused(s.to_string()));
+                return Err(MetadataBuildError::Refused(s.clone()));
+            }
+            // This likely occurs when there is a reorg as well,
+            // but it could also be because validators are colluding
+            if let Err(MetadataBuildError::MerkleRootMismatch {
```

Comment:
I'm assuming we want to stop building the metadata if there are any indication of a merkle root mismatch.